seesaw/neopixel/neopixel.go

```diff
@@ -0,0 +1,181 @@
+package neopixel
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+	"tinygo.org/x/drivers/seesaw"
+)
+
+// seesawWriteDelay the seesaw is quite timing sensitive and times out if not given enough time,
+// this is an empirically determined delay that seems to have good results
+const seesawWriteDelay = time.Millisecond * 10
+
+// RGBW represents the RGBW color of an LED.
+type RGBW struct {
+	R, G, B, W uint8
+}
+
+type Device struct {
+	seesaw          *seesaw.Device
+	ledCount        int
+	pin             uint8
+	pixelType       PixelType
+	lastOperationAt time.Time
+}
+
+func New(dev *seesaw.Device, pin uint8, ledCount int, pixelType PixelType) (*Device, error) {
+
+	pixel := &Device{
+		seesaw:    dev,
+		ledCount:  ledCount,
+		pin:       pin,
+		pixelType: pixelType,
+	}
+
+	if !pixel.checkBufferLength(ledCount) {
+		return nil, fmt.Errorf("invalid LED count: %d", ledCount)
+	}
+
+	time.Sleep(seesawWriteDelay)
+
+	err := pixel.setupPin()
+	if err != nil {
+		return nil, fmt.Errorf("failed to update NeoPixel pin %d: %w", pin, err)
+	}
+
+	time.Sleep(seesawWriteDelay)
+
+	err = pixel.setupLedCount()
+	if err != nil {
+		return nil, fmt.Errorf("failed to update LED count %d: %w", ledCount, err)
+	}
+
+	time.Sleep(seesawWriteDelay)
+
+	err = pixel.setupSpeed()
+	if err != nil {
+		return nil, fmt.Errorf("failed to update pixel type: %w", err)
+	}
+
+	time.Sleep(seesawWriteDelay)
+
+	return pixel, nil
+}
+
+func (s *Device) setupLedCount() error {
+
+	lenBytes := calculateBufferLength(s.ledCount, s.pixelType)
+	buf := []byte{byte(lenBytes >> 8), byte(lenBytes & 0xFF)}
+	return s.seesaw.Write(seesaw.ModuleNeoPixelBase, seesaw.FunctionNeopixelBufLength, buf)
+}
+
+func calculateBufferLength(ledCount int, pixelType PixelType) int {
+	return ledCount * pixelType.EncodedLen()
+}
+
+func (s *Device) setupSpeed() error {
+	speed := byte(0)
+	if s.pixelType.Is800KHz() {
+		speed = 1
+	}
+	return s.seesaw.WriteRegister(seesaw.ModuleNeoPixelBase, seesaw.FunctionNeopixelSpeed, speed)
+}
+
+func (s *Device) setupPin() error {
+	return s.seesaw.WriteRegister(seesaw.ModuleNeoPixelBase, seesaw.FunctionNeopixelPin, s.pin)
+}
+
+// WriteColorAtOffset updates the color for a single LED at the given offset
+func (s *Device) WriteColorAtOffset(offset uint16, color RGBW) error {
+
+	encodedLen := s.pixelType.EncodedLen()
+
+	buf := make([]byte, encodedLen)
+	l := s.pixelType.PutRGBW(buf, color)
+	if l != encodedLen {
+		panic("unexpected encoded length: " + strconv.Itoa(l) + " != " + strconv.Itoa(encodedLen))
+	}
+	byteOffset := offset * uint16(encodedLen)
+	return s.writeBuffer(byteOffset, buf)
+}
+
+// WriteColors writes the given colors to the seesaws NeoPixel buffer
+func (s *Device) WriteColors(buf []RGBW) error {
```

Comment:
Similar to https://github.com/tinygo-org/drivers/blob/release/ws2812/ws2812.go#L36 but with RGBW (that's what the device supports).